core/src/main/java/io/questdb/cairo/TableConverter.java

```diff
@@ -0,0 +1,148 @@
+
+import io.questdb.cairo.sql.TableRecordMetadata;
+import io.questdb.cairo.vm.Vm;
+import io.questdb.cairo.vm.api.MemoryCMR;
+import io.questdb.cairo.vm.api.MemoryMARW;
+import io.questdb.cairo.wal.WalUtils;
+import io.questdb.cairo.wal.seq.TableSequencerAPI;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.*;
+import io.questdb.std.str.Path;
+import io.questdb.std.str.StringSink;
+
+import static io.questdb.cairo.TableUtils.*;
+import static io.questdb.cairo.wal.WalUtils.CONVERT_FILE_NAME;
+
+class TableConverter {
+    private static final Log LOG = LogFactory.getLog(TableConverter.class);
+
+    static void convertTables(CairoConfiguration configuration, TableSequencerAPI tableSequencerAPI) {
+        final Path path = Path.getThreadLocal(configuration.getRoot());
+        final int rootLen = path.length();
+        final StringSink sink = Misc.getThreadLocalBuilder();
+        final FilesFacade ff = configuration.getFilesFacade();
+        final long findPtr = ff.findFirst(path.$());
+        do {
+            if (ff.isDirOrSoftLinkDirNoDots(path, rootLen, ff.findName(findPtr), ff.findType(findPtr), sink)) {
+                if (!ff.exists(path.concat(WalUtils.CONVERT_FILE_NAME).$())) {
+                    continue;
+                }
+                final String dirName = sink.toString();
+                final boolean walEnabled = readWalEnabled(path, ff);
+                LOG.info().$("Converting table [dirName=").$(dirName).$(", walEnabled=").$(walEnabled).$(']').$();
+
+                path.trimTo(rootLen).concat(dirName);
+                try (final MemoryMARW metaMem = Vm.getMARWInstance()) {
+                    openSmallFile(ff, path, rootLen, metaMem, META_FILE_NAME, MemoryTag.MMAP_SEQUENCER_METADATA);
+                    if (metaMem.getBool(TableUtils.META_OFFSET_WAL_ENABLED) == walEnabled) {
+                        LOG.info().$("Skipping conversion, table already has the expected type [dirName=").$(dirName).$(", walEnabled=").$(walEnabled).$(']').$();
+                    } else {
+                        if (walEnabled) {
+                            final String tableName;
+                            try (final MemoryCMR mem = Vm.getCMRInstance()) {
+                                final String name = TableUtils.readTableName(path, mem, ff);
+                                tableName = name != null ? name : dirName;
+                            }
+
+                            final int tableId = metaMem.getInt(TableUtils.META_OFFSET_TABLE_ID);
+                            final TableToken token = new TableToken(tableName, dirName, tableId, true);
+                            try (TableReaderMetadata metadata = new TableReaderMetadata(configuration, token)) {
+                                metadata.load();
+                                tableSequencerAPI.registerTable(tableId, new TableDescriptorImpl(metadata), token);
+                            }
+                        }
```

Comment:
`if (!walEnabled)` we need to clean sequencer files, otherwise next conversion back to WAL may have problems.
Also I'd rename table dir into `<table name>`

Reply:
left the sequencer files there intentionally.
my thinking was that better to keep it in case they convert back to WAL.
I was thinking about keeping seq txn number and wal ids unique.
if we clear the sequencer files we probably need to clear seq txn number in the table dir too in _meta, otherwise wal apply job will be out of sync.

also skipped the renaming of the table dir, thought why to mess with it if not necessary to do it?

Reply:
will cleanup the sequencer dir + wal dirs too.
that is because the table's schema could change while it is non-WAL and when it becomes WAL again metadata would be out of sync.
going to non-WAL means losing all WAL-related files/history.

Reply:
fixed in [b905162](https://github.com/questdb/questdb/pull/2947/commits/b9051628961a88566baf8489f0e116941e61f4e5)

now when converting to non-WAL:
 - we remove all wal* dirs and lock files
 - we also remove sequencer persistence (txn_seq dir)
 - reset the lastAppliedSeqTxn to 0 in _txn